module/zfs/zio.c

```diff
@@ -5492,10 +5492,11 @@ zio_done(zio_t *zio)
 		    zio->io_error == ENXIO &&
 		    spa_load_state(zio->io_spa) == SPA_LOAD_NONE &&
 		    spa_get_failmode(zio->io_spa) != ZIO_FAILURE_MODE_CONTINUE)
-			zio->io_reexecute |= ZIO_REEXECUTE_SUSPEND;
+			zio->io_post |= ZIO_POST_SUSPEND;
 
-		if (!(zio->io_flags & ZIO_FLAG_CANFAIL) && !zio->io_reexecute)
-			zio->io_reexecute |= ZIO_REEXECUTE_SUSPEND;
+		if (!(zio->io_flags & ZIO_FLAG_CANFAIL) &&
+		    !(zio->io_post & (ZIO_POST_REEXECUTE|ZIO_POST_SUSPEND)))
+			zio->io_post |= ZIO_POST_REEXECUTE;
```

Comment:
Was this functional change intended.  Shouldn't this be `ZIO_POST_SUSPEND`?

Reply:
Sigh, that's embarrassing. Well spotted on both. My guess, something was lost on a series of renames - these weren't their final form.

Fixed both, and I re-read the diff out-loud to make sure they match up. Which probably just means I've stuffed it in a new way...